order-book/src/main/java/com/market/banica/order/book/service/grpc/OrderBookService.java

```diff
@@ -64,35 +47,37 @@ public void getOrderBookItemLayers(ItemOrderBookRequest request, StreamObserver<
 
     @Override
     public void announceItemInterest(InterestsRequest request, StreamObserver<InterestsResponse> responseObserver) {
-
+        String itemName = request.getItemName();
         try {
 
-            auroraClient.startSubscription(request.getItemName(), request.getClientId());
+            auroraClient.startSubscription(itemName, request.getClientId());
             responseObserver.onNext(InterestsResponse.newBuilder().build());
             responseObserver.onCompleted();
             LOGGER.info("Announce item interest by client id: {}", request.getClientId());
 
         } catch (TrackingException e) {
 
-            LOGGER.warn("Announce item interest by client id: {} has failed with item: {}", request.getClientId(), request.getItemName());
+            LOGGER.warn("Announce item interest by client id: {} has failed with item: {}",
+                    request.getClientId(), itemName);
             responseObserver.onError(Status.INVALID_ARGUMENT.withDescription("Invalid item name").asException());
 
         }
     }
 
     @Override
     public void cancelItemSubscription(CancelSubscriptionRequest request, StreamObserver<CancelSubscriptionResponse> responseObserver) {
+        String itemName = request.getItemName();
 
         try {
 
-            auroraClient.stopSubscription(request.getItemName(), request.getClientId());
+            auroraClient.stopSubscription(itemName, request.getClientId());
```

Comment:
if you extracted the item name because it is repeated twice, you should do it also for the clientId

Reply:
Fixed